pre_processing/read_slstr_funcs.F90

```diff
@@ -229,13 +233,31 @@ subroutine read_slstr_visdata(indir, inband, outarr, imager_angles, &
    ! Find the filename required for this channel
    call get_slstr_imnames(indir, inband, filename, filename_qa, bandname, irradname)
 
-   allocate(data1(nx*2,ny*2))
+   ! Open the netcdf file
+   call ncdf_open(fid, filename, 'read_slstr_visdata()')
+   ! Determin the actual size of the data array (which should be
+   ! nx*2,ny*2, but sometimes isn't). Ensure the data array is at
+   ! least nx*2, ny*2 in size
+   ny1 = ncdf_dim_length(fid, 'rows', 'read_slstr_visdata()')
+   nx1 = ncdf_dim_length(fid, 'columns', 'read_slstr_visdata()')
+   if (ny1 .gt. ny*2) then
+      nyr = ny1
+   else
+      nyr = ny*2
+   end if
+   if (nx1 .gt. nx*2) then
+      nxr = nx1
+   else
+      nxr = nx*2
+   end if
+
+   allocate(data1(nxr,nyr))
 
    data1(:,:) = sreal_fill_value
 
    ! Open the netcdf file
-   call ncdf_open(fid, filename, 'read_slstr_visdata()')
-   call ncdf_read_array(fid, bandname, data1, start=[sx*2-1, sy*2-1])
+   call ncdf_read_array(fid, bandname, data1(1:nx1,1:ny1), &
```

Comment:
Ok, I think I've come up with the workable solution. As mentioned below, although these frames have one extra row of VIS data, they also have one extra row of IR data. Therefore, the problem isn't that the VIS data has one more row  than the code is expecting, but that they have one _LESS_.
I've changed my code so that it deals with this specific issue and allows for exactly one fewer than nx*2 or ny*2 columns or rows in the VIS grid. Any more than that, and it will still fall over. It should also handle the sx and sy parameters.

Reply:
Cool. Push it up and I'll give it a try

Reply:
It should be there...